src/schema/v2/homeView/artworkResolvers.ts

```diff
@@ -0,0 +1,92 @@
+import type { GraphQLFieldResolver } from "graphql"
+import type { HomeViewSectionKey } from "./getSectionsForUser"
+import type { ResolverContext } from "types/graphql"
+import { artworksForUser } from "../artworksForUser"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { paginationResolver } from "../fields/pagination"
+
+/*
+ * Resolvers for home view artwork sections
+ */
+
+// the resolvers
+
+const newWorksForYouResolver: GraphQLFieldResolver<
+  any,
+  ResolverContext
+> = async (parent, args, context, info) => {
+  const finalArgs = {
+    // formerly specified client-side
+    maxWorksPerArtist: 3,
+    includeBackfill: true,
+    first: args.first,
+    version: "C",
+    excludeDislikedArtworks: true,
+    excludeArtworkIds: [],
+
+    ...args,
+  }
```

Comment:
The big paradigm shift here is that the _server_ becomes authority for deciding what artworks should be selected for any given artworks rail.

This this kind of filtering logic, which previously was specified in the client…

https://github.com/artsy/eigen/blob/323364090372beec0d2a0bf4555795e7336c4def/src/app/Components/WorksForYouArtworks.tsx#L69-L73

…will now move up to the server, i.e. L20–24 above.

Reply:
L25 is a hacky way of solving the requirement that certain args be specified with default values. I believe @nickskalkin suggested  a helper (e.g. something like `extractDefaultsFromField` that could introspect over a graphql field config and pull out [any default values that have been defined](https://github.com/artsy/metaphysics/blob/d160f6629aeeecf4089cd652a0c433b87af061be/src/schema/v2/artworksForUser/artworksForUser.ts#L34-L45))